Implemented Dark Mode

edge33 opened the pull request:
Features:

Added dark mode toggle and CSS:

https://github.com/edge33/plainwhite-jekyll/blob/master/dark.png

thelehhman commented:
Wow that's great! Thanks @edge33.
A couple of things to note:
- I think yellow & orange don't align well with the overall theme, its better to use grayish colors instead.
- This has a couple of extra files which would hamper the merge, do you mind taking care of that?
- README has dark mode at top. It is undesirable there. It should be somewhere down with its own heading.

edge33 commented:
Hey @thelehhman, I am glad you appreciated:
- Do you have any preferences for the colors?
- What file are you referring to? the screenshots ? how should I manage those?
- Right, I'll fix that

thelehhman commented:
1. Something not much contrasting. Shades of gray/white preferably.
2. code of conduct, email address.png etc.

Also, it would be better if dark mode could be enabled optionally by setting it in `_config.yml`. See https://github.com/thelehhman/plainwhite-jekyll/pull/21 for example, it enables the nav by an option.

edge33 commented:
HI @thelehhman, I updated the code, see if you like the theme better now, I also added configuraton for conditional activation and added the feature that reads the OS theme to set the website accordingly.

I wanted to remove the toggle and use an icon with the 'moon' like the ones you are using for the social buttons, but I was not able to do so, what would you think about having this icon instead of the toggle?

thelehhman commented on README.md:
```diff
@@ -1,5 +1,3 @@
-# plainwhite
-
```
We don't want to remove these lines.

thelehhman commented on assets/css/style.scss:
```diff
@@ -1,4 +1,6 @@
-@import "plain";
\ No newline at end of file
+@import "plain";
+@import "toggle";
+@import "dark";
```
We could wrap this in a if block with `site.plainwhite.dark_mode` so that we load only the css which is required.

thelehhman commented on assets/js/toggle.js:
```diff
@@ -0,0 +1,36 @@
+function toggleDarkMode() {
+    const DARK_CLASS = 'dark';
+
+    var body = document.querySelector("body");
+    if (body.classList.contains(DARK_CLASS)) {
+        setCookie('theme', 'light');
+        body.classList.remove(DARK_CLASS);
+    } else {
+        setCookie('theme', 'dark');
+        body.classList.add(DARK_CLASS);
+    }
+}
+
+function getCookie(name) {
+    var v = document.cookie.match('(^|;) ?' + name + '=([^;]*)(;|$)');
+    return v ? v[2] : null;
+}
+function setCookie(name, value, days) {
+    var d = new Date;
+    d.setTime(d.getTime() + 24*60*60*1000*days);
+    document.cookie = name + "=" + value + ";path=/;expires=" + d.toGMTString();
+}
+
+function deleteCookie(name) { setCookie(name, '', -1); }
```
I think we are better of adding a separate file for this, `darkmode.js` so that we don't load it if we don't need to.

edge33 replied:
Yeah! you're definitely right, I din't think about that! I'll put the script in the if block, so it won't get loaded if dark mode is disabled!

thelehhman commented:
Great going! Just address the issues mentioned above and we'll be good to go :')

edge33 commented:
Hi @thelehhman, I addressed the issues, can you please check?

thelehhman commented:
It looks great. The last thing we need to do is to squash all your commits into one. This helps to maintain a cleaner history.

edge33 commented:
and that is done @thelehhman

thelehhman commented:
Lovely, thanks a lot!
